adhocteam/Head-Start-TTADP
# Edit goals without losing objectives

## Description of change

Added the ability to edit Goal names from the goals-objectives section of an activity report. I removed the trash icon button and added a ContextMenu with options to 'Edit' or 'Remove' a goal (the remove functionality being provided by the trash button before).

The 'Edit' button swaps the goal name with an InplaceGoalEditor, which is essentially an input with save and cancel buttons. I added two props to the Goal component: `onUpdateGoal` and `isEditable` (in case we need to use the component to display non-editable goals). The GoalPicker component has an `onUpdateGoal` handler that manages updating the selected goals on the front end.

On the back end, I updated the `saveGoalsForReport` function in src/services/goals.js to be able to update goals. Before, only goals with a null/non-numeric id would be saved. 

In addition, I added an order prop to the query for `activityReportById` so we'd get objectives by id (ascending). I was seeing goals display in a different order on refresh, which looked odd, and this mitigates that.

## How to test

Create an activity report, add some goals and objectives, then save the report. Go back to the goals/objectives and click the context menu for a goal, and select 'edit'. Change the goal name, click 'Update Goal', then click Save. The goal should be updated in the database.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-87


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
